### PR TITLE
Add warning tests of integer vectors

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

 

resolves https://github.com/pyvista/pyvista/issues/2660

### Details

- None

### CODE REVIEW

File: tests/test_filters.py [hunk 1]
```diff
@@ -2180,6 +2180,20 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
...
+    for dataset in datasets:
+        tf = pyvista.transformations.axis_angle_rotation((1, 0, 0), 90)
+        dataset.clear_data()
+        for i in range(num_cell_arrays):
+            dataset.cell_data['C%d' % i] = np.random.randint(0, sys.maxsize, (dataset.n_cells, 3))
+        for i in range(num_point_data):
+            dataset.point_data['P%d' % i] = np.random.randint(0, sys.maxsize, (dataset.n_points, 3))
+        if not (num_cell_arrays == 0 and num_point_data == 0):
+            with pytest.warns(UserWarning, match="Integer"):
+                transformed = dataset.transform(tf, transform_all_input_vectors=True, inplace=False)
```
Comment (tkoyama010) [inline tests/test_filters.py:2194]:
```suggestion
                _ = dataset.transform(tf, transform_all_input_vectors=True, inplace=False)
```

File: tests/test_filters.py [hunk 2]
```diff
@@ -2180,6 +2180,20 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
             )
 
 
+@pytest.mark.parametrize('num_cell_arrays,num_point_data', itertools.product([0, 1, 2], [0, 1, 2]))
+def test_transform_int_vectors_warning(datasets, num_cell_arrays, num_point_data):
+    for dataset in datasets:
+        tf = pyvista.transformations.axis_angle_rotation((1, 0, 0), 90)
+        dataset.clear_data()
+        for i in range(num_cell_arrays):
+            dataset.cell_data['C%d' % i] = np.random.randint(0, sys.maxsize, (dataset.n_cells, 3))
```
Comment (tkoyama010) [inline tests/test_filters.py:2189]:
```suggestion
            dataset.cell_data['C%d' % i] = np.random.randint(0, size=(dataset.n_cells, 3))
```

File: tests/test_filters.py [hunk 3]
```diff
@@ -2180,6 +2180,20 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
...
+        for i in range(num_point_data):
+            dataset.point_data['P%d' % i] = np.random.randint(0, sys.maxsize, (dataset.n_points, 3))
```
Comment (tkoyama010) [inline tests/test_filters.py:2191]:
```suggestion
            dataset.point_data['P%d' % i] = np.random.randint(0, size=(dataset.n_points, 3))
```

File: tests/test_filters.py [hunk 4]
```diff
@@ -2180,6 +2180,20 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
             )
 
 
+@pytest.mark.parametrize('num_cell_arrays,num_point_data', itertools.product([0, 1, 2], [0, 1, 2]))
+def test_transform_int_vectors_warning(datasets, num_cell_arrays, num_point_data):
+    for dataset in datasets:
+        tf = pyvista.transformations.axis_angle_rotation((1, 0, 0), 90)
+        dataset.clear_data()
+        for i in range(num_cell_arrays):
+            dataset.cell_data['C%d' % i] = np.random.randint(0, size=(dataset.n_cells, 3))
```
Comment (tkoyama010) [inline tests/test_filters.py:2189]:
```suggestion
            dataset.cell_data['C%d' % i] = np.random.randint(np.iinfo(int).max, size=(dataset.n_cells, 3))
```

File: tests/test_filters.py [hunk 5]
```diff
@@ -2180,6 +2180,20 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
...
+        for i in range(num_point_data):
+            dataset.point_data['P%d' % i] = np.random.randint(0, size=(dataset.n_points, 3))
```
Comment (tkoyama010) [inline tests/test_filters.py:2191]:
```suggestion
            dataset.point_data['P%d' % i] = np.random.randint(np.iinfo(int).max, size=(dataset.n_points, 3))
```

File: tests/test_filters.py [hunk 6]
```diff
@@ -2180,6 +2180,29 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
             )
 
 
+
```
Comment (tkoyama010) [inline tests/test_filters.py:2183]:
```suggestion
```

File: tests/test_filters.py [hunk 7]
```diff
@@ -2180,6 +2180,24 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
             )
 
 
+@pytest.mark.parametrize("num_cell_arrays,num_point_data", itertools.product([0, 1, 2], [0, 1, 2]))
+def test_transform_int_vectors_warning(datasets, num_cell_arrays, num_point_data):
```
Comment (MatthewFlamm) [inline tests/test_filters.py:2184]: Can we also get a test that makes sure that cell vectors are transformed correctly when not an integer?  It seems the test above this one is broken.  I assume that the transform is overwriting the data in the original mesh, which could be another bug.  I haven't delved too deeply though.

Comment (tkoyama010) [reply tests/test_filters.py:2184]: Good point, I checked `test_transform_mesh_and_vectors` and indeed the checks are not working. I've determined the `test_transform_mesh_and_vectors` source and it seems to be fine. I can't figure out the cause right now. I think it needs to be investigated.

Comment (akaszynski) [reply tests/test_filters.py:2184]: Do we need this to be resolved before reviewing/merging this?

Comment (tkoyama010) [reply tests/test_filters.py:2184]: Yes. Resolving #2660 entirely is the goal of this PR.

Comment (akaszynski) [reply tests/test_filters.py:2223]: > I assume that the transform is overwriting the data in the original mesh, which could be another bug.

Added an additional check for this with 6a223884b80dd72f3c60370a5ba8ac2438fe5e6c.

> Can we also get a test that makes sure that cell vectors are transformed correctly when not an integer?

This is being checked and the test is valid. I like how the test is written to validate a 90 degree rotation.
https://github.com/pyvista/pyvista/blob/6a223884b80dd72f3c60370a5ba8ac2438fe5e6c/tests/test_filters.py#L2199-L2208

**Comment by akaszynski** [comment]
Comment https://github.com/pyvista/pyvista/pull/2664#discussion_r888291883 should be resolved as of [6a22388](https://github.com/pyvista/pyvista/pull/2664/commits/6a223884b80dd72f3c60370a5ba8ac2438fe5e6c).

**Review by akaszynski — APPROVED** [review]

**Pull request merged**